server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -47,9 +53,52 @@ public Invocations getInvocations() {
             .invocation().methods(GET).path("/v2/server/stop").handleWith(this::stop)
             .invocation().methods(GET).path("/v2/server/threads").handleWith(this::threads)
             .invocation().methods(GET).path("/v2/server/cache-managers").handleWith(this::cacheManagers)
+            .invocation().methods(GET).path("/v2/server/ignored-caches/{cache-manager}").handleWith(this::listIgnored)
+            .invocation().methods(POST, DELETE).path("/v2/server/ignored-caches/{cache-manager}/{cache}").handleWith(this::doIgnoreOp)
             .create();
    }
 
+   private CompletionStage<RestResponse> doIgnoreOp(RestRequest restRequest) {
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      boolean add = restRequest.method().equals(POST);
+
+      String cacheManagerName = restRequest.variables().get("cache-manager");
+      DefaultCacheManager cacheManager = invocationHelper.getServer().getCacheManager(cacheManagerName);
+
+      if (cacheManager == null) return completedFuture(builder.status(NOT_FOUND).build());
+
+      String cacheName = restRequest.variables().get("cache");
+
+      if (!cacheManager.getCacheNames().contains(cacheName)) {
+         return completedFuture(builder.status(NOT_FOUND).build());
+      }
+      ServerManagement server = invocationHelper.getServer();
+
+      if (add) {
+         return server.ignoreCache(cacheManagerName, cacheName).thenApply(r -> builder.build());
+      } else {
+         return server.unIgnoreCache(cacheManagerName, cacheName).thenApply(r -> builder.build());
```

Comment:
* if the `CompletableFuture` completes exceptionally, is the exception handled somewhere?

Reply:
All REST resources completion errors are handled at https://github.com/infinispan/infinispan/blob/06995a31d242c4195f8100a688ac6602cdbb48bd/server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java#L134